server/sonar-web/src/main/js/components/charts/LanguageDistribution.tsx

```diff
@@ -28,34 +28,41 @@ interface Props {
   alignTicks?: boolean;
   distribution: string;
   languages?: Language[];
+  width: number;
 }
 
 export default function LanguageDistribution(props: Props) {
-  let data = props.distribution.split(';').map((point, index) => {
+  let distribution = props.distribution.split(';').map(point => {
     const tokens = point.split('=');
-    return { x: parseInt(tokens[1], 10), y: index, value: tokens[0] };
+    return { language: tokens[0], lines: parseInt(tokens[1], 10) };
   });
 
-  data = sortBy(data, d => -d.x);
+  distribution = sortBy(distribution, d => -d.lines);
 
-  const yTicks = data.map(point => getLanguageName(point.value)).map(cutLanguageName);
-  const yValues = data.map(point => formatMeasure(point.x, 'SHORT_INT'));
+  const data = distribution.map(d => d.lines);
+  const yTicks = distribution.map(d => getLanguageName(d.language)).map(cutLanguageName);
+  const yTooltips = distribution.map(d => (d.lines > 1000 ? formatMeasure(d.lines, 'INT') : ''));
+  const yValues = distribution.map(d => formatMeasure(d.lines, 'SHORT_INT'));
 
   return (
     <Histogram
       alignTicks={props.alignTicks}
-      data={data}
+      bars={data}
+      height={distribution.length * 25}
+      padding={[0, 60, 0, 80]}
       yTicks={yTicks}
+      yTooltips={yTooltips}
       yValues={yValues}
-      barsWidth={10}
-      height={data.length * 25}
-      padding={[0, 60, 0, 80]}
+      width={props.width}
     />
   );
 
   function getLanguageName(langKey: string) {
```

Comment:
Didn't you want to put those two functions after the component ? It's currently inside.

Reply:
`getLanguageName` needs access to the component props, so it's better to keep it inside, but `cutLanguageName` can be moved outside